pallets/author-mapping/src/lib.rs

```diff
@@ -283,4 +283,40 @@ pub mod pallet {
 			Self::account_and_deposit_of(author_id).map(|info| info.account)
 		}
 	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_runtime_upgrade() -> Weight {
+			let pallet_prefix: &[u8] = b"AuthorMapping";
+			let old_storage_item_prefix: &[u8] = b"MappingWithDeposit";
+
+			let mut migrated_entries: Weight = 0;
+
+			// Move the mappings to the new storage location
+			// https://crates.parity.io/frame_support/storage/migration/fn.storage_key_iter.html
+			for (author_id, account_id) in frame_support::storage::migration::storage_key_iter::<
+				T::AuthorId,
+				RegistrationInfo<T::AccountId, BalanceOf<T>>,
+				Twox64Concat,
+			>(pallet_prefix, old_storage_item_prefix)
+			{
+				Mapping::<T>::insert(author_id, account_id);
+				migrated_entries += 1;
+			}
+
+			// Now remove the old storage
+			// https://crates.parity.io/frame_support/storage/migration/fn.remove_storage_prefix.html
+			frame_support::storage::migration::remove_storage_prefix(
+				pallet_prefix,
+				old_storage_item_prefix,
+				&[],
```

Comment:
maybe a dumb question, why is this just empty? when would it not be empty? I looked at the source and don't really understand what this last parameter is.

Reply:
The doc comment is incorrect too, if anyone wants an easy substrate PR lol

Reply:
This comes from `clear_prefix` which this fn eventually chains to:
```rust
/// Clear the storage of each key-value pair where the key starts with the given `prefix`
```

Reply:
If the item in question is a map, and you want to only delete one entry in the map, you could supply the hashed key.

Or if ti is a double map and you wanted to remove by first key, you could supply just that part.